### ISSUE TITLE
Some configuration options don't respect --config setting

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#288]
The configuration file generated by running `blockstack -y --config config/client.ini` contains some values that use the default `~/.blockstack` config directory instead of the directory specified by `--config`

```
(venv) Vanquish:blockstack-core larry$ cat config/client.ini 
[blockstack-client]
datastores = datastores
default_devices = 
accounts = accounts
queue_path = /Users/larry/.blockstack/queues.db
blockchain_reader = blockcypher
rpc_detach = True
advanced_mode = False
authenticate_api = True
port = 6264
email = 
blockchain_headers = /Users/larry/.blockstack/blockchain-headers.dat
storage_drivers_local = disk
users = accounts
rpc_token = 93b2c2fb5a459442a984449151e8d69a44129a6b7f5dad34e5d9b62eb1d51c5b
anonymous_statistics = True
blockchain_writer = blockcypher
poll_interval = 300
server = node.blockstack.org
storage_drivers_required_write = disk,blockstack_server,dht
api_endpoint_port = 6270
storage_drivers = disk,blockstack_resolver,blockstack_server,http,dht
client_version = 0.14.1.0
metadata = /Users/larry/.blockstack/metadata

[blockchain-reader]
utxo_provider = blockcypher
api_token = 

[bitcoind]
passwd = blockstacksystem
regtest = False
spv_path = /Users/larry/.virtualchain-spv-headers.dat
server = bitcoin.blockstack.com
p2p_port = 8333
user = blockstack
timeout = 300.0
port = 8332

[blockchain-writer]
utxo_provider = blockcypher
api_token = 

```

**Comment by jcnelson** [comment]
Thanks for this!

Regarding `spv_path` in `[bitcoind]`, this is set by `virtualchain`, not by Blockstack.  Blockstack's API daemon, its indexer, and Virtualchain all share this file.  This is why it's outside `~/.blockstack`.  In the future, this particular field should be set in some global config file like `/etc/virtualchain.conf` or the like.

I'll remove `blockchain_headers` from `[blockstack-client]`.  That file stores the exact same data as `~/.virtualchain-spv-headers.dat`.  The fact that they're separate is an artifact of legacy code (i.e. before Virtualchain had an SPV client).

The rest of the paths that start with `/Users/larry` should reflect your choice of config file path with `--config`.  I'll fix those.

**Comment by jcnelson** [comment]
Pushed fixes to rc-0.14.1b.

**Comment by larrysalibra** [comment]
This works for me now! Thank you!
